src/m365/teams/commands/chat/chat-get.ts

```diff
@@ -0,0 +1,200 @@
+    const currentUserEmail = accessToken.getUserNameFromAccessToken(Auth.service.accessTokens[this.resource].accessToken).toLowerCase();
+    const existingChats = await this.findExistingChatsByParticipants([currentUserEmail, ...participants]);
+    
+    if (!existingChats || existingChats.length === 0) {
+      throw new Error('No chat conversation was found with these participants.');
+    }
+
+    if (existingChats.length === 1) {
+      return existingChats[0].id as string;
+    }
+
+    const disambiguationText = existingChats.map(c => {
+      return `- ${c.id}${c.topic && ' - '}${c.topic} - ${c.createdDateTime && new Date(c.createdDateTime).toLocaleString()}`;
+    }).join(os.EOL);
+
+    throw new Error(`Multiple chat conversations with this topic found. Please disambiguate:${os.EOL}${disambiguationText}`);
+    
+  }
+  
+  private async getChatIdByName(name: string): Promise<string> {
+    const existingChats = await this.findExistingGroupChatsByTopic(name);
+
+    if (!existingChats || existingChats.length === 0) {
+      throw new Error('No chat conversation was found with this name.');
+    }
+
+    if (existingChats.length === 1) {
+      return existingChats[0].id as string;
+    }
+
+    const disambiguationText = existingChats.map(c => {
+      const memberstring = (c.members as ConversationMember[]).map(m => (m as AadUserConversationMember).email).join(', ');
+      return `- ${c.id} - ${c.createdDateTime && new Date(c.createdDateTime).toLocaleString()} - ${memberstring}`;
+    }).join(os.EOL);
+
+    throw new Error(`Multiple chat conversations with this topic found. Please disambiguate:${os.EOL}${disambiguationText}`);
+  }
+  
+  private async findExistingChatsByParticipants(expectedMemberEmails: string[]): Promise<Chat[]> {
+    const chatType = expectedMemberEmails.length === 2 ? "oneOnOne" : "group";
+    const endpoint = `${this.resource}/v1.0/chats?$filter=chatType eq '${chatType}'&$expand=members&$select=id,topic,createdDateTime,members`;
+    const foundChats: Chat[] = [];
+
+    const chats = await this.getAllChats(endpoint, []);
+
+    for (const chat of chats) {
+      const chatMembers = chat.members as ConversationMember[];
+      if (chatMembers.length === expectedMemberEmails.length) {
+        const chatMemberEmails = chatMembers.map(member => (member as AadUserConversationMember).email?.toLowerCase());
+
+        if (expectedMemberEmails.every(email => chatMemberEmails.some(memberEmail => memberEmail === email))) {
+          foundChats.push(chat);
+        }
+      }
+    }
+
+    return foundChats;
+  }
+
+  private async findExistingGroupChatsByTopic(topic: string): Promise<Chat[]> {
+    const endpoint = `${this.resource}/v1.0/chats?$filter=topic eq '${encodeURIComponent(topic)}'&$expand=members&$select=id,topic,createdDateTime,chatType`;
+    const chats = await this.getAllChats(endpoint, []);
```

Comment:
I don't think we need to `await` the results here and could directly return all chats